pkg/substitution/substitution.go

```diff
@@ -14,27 +15,27 @@ const (
 	cloudNodeManagerName       = "cloud-node-manager"
 )
 
-// setDeploymentImages substitutes controller containers in Deployment with correct image
-func setDeploymentImages(config config.OperatorConfig, d *v1.Deployment) {
-	for i, container := range d.Spec.Template.Spec.Containers {
-		if container.Name != cloudControllerManagerName {
+// setCloudControllerImage substitutes controller containers in provided pod specs with correct image
+func setCloudControllerImage(config config.OperatorConfig, p corev1.PodSpec) corev1.PodSpec {
+	updatedPod := *p.DeepCopy()
+	for i, container := range p.Containers {
+		substitute := ""
```

Comment:
Bit of a nit, but, would it be better to call this `substituteName`, would be less ambiguous IMO because substitute could be a verb of noun but we definitely want a noun here